lib/Modules/ComputeHooks/ComputeHooks.psm1

```diff
@@ -106,1360 +56,800 @@ function Install-Prerequisites {
+                    "mandatory" = $true
                 },
                 @{
-                    "generator"="Get-GlanceContext";
-                    "relation"="image-service";
+                    "generator" = (Get-Item "function:Get-GlanceContext").ScriptBlock
+                    "relation" = "image-service"
+                    "mandatory" = $true
                 },
                 @{
-                    "generator"="Get-CharmConfigContext";
-                    "relation"="config";
+                    "generator" = (Get-Item "function:Get-CharmConfigContext").ScriptBlock
+                    "relation" = "config"
+                    "mandatory" = $true
                 },
                 @{
-                    "generator"="Get-SystemContext";
-                    "relation"="system";
+                    "generator" = (Get-Item "function:Get-SystemContext").ScriptBlock
+                    "relation" = "system"
+                    "mandatory" = $true
                 },
                 @{
-                    "generator"="Get-S2DContext";
-                    "relation"="s2d";
-                }
-            );
-        };
-        "neutron"=@{
-            "myname"="neutron";
-            "template"="$template_dir\$distro\neutron_hyperv_agent.conf"
-            "service"="neutron-hyperv-agent";
-            "binpath"="$neutronHypervAgentExe";
-            "serviceBinPath"="`"$serviceWrapperNeutron`" neutron-hyperv-agent `"$neutronHypervAgentExe`" --config-file `"$neutron_config`"";
-            "config"="$neutron_config";
-            "context_generators"=@(
-                @{
-                    "generator"="Get-RabbitMQContext";
-                    "relation"="amqp";
+                    "generator" = (Get-Item "function:Get-S2DContext").ScriptBlock
+                    "relation" = "s2d"
+                    "mandatory" = $false
                 },
                 @{
-                    "generator"="Get-NeutronContext";
-                    "relation"="cloud-compute";
-                },
+                    "generator" = (Get-Item "function:Get-FreeRDPContext").ScriptBlock
+                    "relation" = "free-rdp"
+                    "mandatory" = $false
+                }
+            )
+        }
+
+        "neutron" = @{
+            "template" = "$distro\neutron_hyperv_agent.conf"
+            "service" = "neutron-hyperv-agent"
+            "binpath" = "$neutronHypervAgentExe"
+            "serviceBinPath" = "`"$serviceWrapperNeutron`" neutron-hyperv-agent `"$neutronHypervAgentExe`" --config-file `"$neutronHypervConf`""
+            "config" = "$neutronHypervConf"
+            "display_name" = "Neutron Hyper-V Agent"
+            "context_generators" = @(
                 @{
-                    "generator"="Get-CharmConfigContext";
-                    "relation"="config";
+                    "generator" = (Get-Item "function:Get-RabbitMQContext").ScriptBlock
```

Comment:
Yep. Moreover, you can pass `-ComputerName` (the name of an AD computer) and `-ArgumentList` (function parameters) and you execute a local function with desired parameters on a remote comptuer.